app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -100,7 +100,11 @@ def parse_cluster(cluster)
       }
       set_relationship_on_hosts(new_result, nodes)
 
-      return uid, new_result
+      if @data[:hosts].select { |host| host[:ems_cluster] == new_result }.empty?
```

Comment:
I had to go back and read the PR description to understand why this filtering was being done.

I'd suggest adding a comment above saying why you have to filter clusters (`# ignore clusters that are left without any hosts after hosts were filtered for UNSUPPORTED_HOST_PLATFORMS`)